TensorLib/Index.lean

```diff
@@ -264,38 +143,6 @@ private def applyWithCopy (index : NumpyBasic) (arr : Tensor) : Err Tensor := do
      data
   }
 
-/-
-Make an attempt to not copy. For example, a partial index into an array
-should probably just return a view with the same (sub) strides.
-This is currently a feeble attempt, but can get fancier over time.
-We currently handle
-
-1. Non-slice indices (e.g. arr[1][7][-2])
-2. TODO
--/
-def apply (index : NumpyBasic) (arr : Tensor) : Err Tensor := do
```

Comment:
I removed this bc it adds complexity, and is incompatible with stuffing nat indicies into singleton slices, which is how I simplified the old "Basic" iterator without needing a "sum" iterator. I may add that if we have another example, but this workaround seemed nice for code simplicity.